qiskit/synthesis/unitary/qsd.py

```diff
@@ -251,14 +251,13 @@ def _get_ucry_cz(nqubits, angles):
 
 
 def _apply_a2(circ):
-    from qiskit.compiler import transpile
     from qiskit.quantum_info import Operator
     from qiskit.circuit.library.generalized_gates.unitary import UnitaryGate
+    from qiskit.transpiler.passes.synthesis import HighLevelSynthesis
 
     decomposer = two_qubit_decompose_up_to_diagonal
-    ccirc = transpile(
-        circ, basis_gates=["u", "cx", "qsd2q"], optimization_level=0, qubits_initially_zero=False
-    )
+    hls = HighLevelSynthesis(basis_gates=["u", "cx", "qsd2q"], qubits_initially_zero=False)
+    ccirc = hls(circ)
```

Comment:
Previously `transpile` built a flow containing `UnitarySynthesis`, `HighLevelSynthesis` and `BasisTranslator`. Looking at `qs_decomposition` that this change is safe in the sense the `HighLevelSynthesis` should be sufficient. Can you just confirm?